cmd/plugin/builder/helpers/parallel.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package helpers
+
+import "runtime"
+
+var minConcurrent = 2
+
+// ErrInfo is used to return error information for
+type ErrInfo struct {
+	Err  error
+	Path string
+	ID   string
+}
+
+// GetMaxParallelism return the maximum concurrent threads to use.
+// Limit the number of concurrent operations we perform so we don't overwhelm the system.
+func GetMaxParallelism() int {
+	maxConcurrent := runtime.NumCPU() - 2
```

Comment:
any specific reason for reducing 2 from the number of cpu count?

Reply:
As mentioned in the comment above, this is to make sure we do not overwhelm the system by using all CPUs to build plugins and leave a few CPUs available to do other operations. This is a prior art where we build plugins in parallel and I am using the same functions to build plugin packages and publishing those in parallel.